components/TravelMethodButtons/TravelMethodButtons.jsx

```diff
@@ -0,0 +1,50 @@
+import { Box, Button, Text, SimpleGrid, Icon, Flex } from '@chakra-ui/react';
+
+export function TravelMethodButtons({
+  transportIcon,
+  modesOfTransport,
+  handleTransportMode,
+  transportMode,
```

Comment:
do we need to pass in these values as parameters to this function?
- `transportIcon` - this is a list of transport icons from `utils/constants.js` - why can't we `import` the icons here instead?
-  `modesOfTransport` - this is a list of accepted transport methods from `utils/constants.js`. Why can't we `import` this directly?
- `handleTransportMode` - this is the click event handler for the buttons. I think it makes sense to define the `TravelMethodButtons` button click event handler within the `TravelMethodButtonsContainer` itself... but there are pros for passing it in as a parameter too. What do you think @batteries76 , @OlaShabalina, @ymloh ?
- `transportMode` - this is a default value for a travel method, Which we don't really need. Under what circumstances would a survey offer 'default' answers for questions? Does it apply to the survey we are building?

```suggestion
```

Reply:
hgtan question > transportIcon - this is a list of transport icons from utils/constants.js - why can't we import the icons here instead?
modesOfTransport - this is a list of accepted transport methods from utils/constants.js. Why can't we import this directly?

Answer > These icons are used in array form together with the text are displayed with the array of name listed in constant.jsx. This code to retrieve icons sit well with the map method used for retrieving the button text.
I hope this answer your question.

question > handleTransportMode - this is the click event handler for the buttons. I think it makes sense to define the TravelMethodButtons button click event handler within the TravelMethodButtonsContainer itself... but there are pros for passing it in as a parameter too. What do you think @batteries76 , @OlaShabalina, @ymloh ?

Answer > Currently the I'm focussing only on design. I haven't change the functionality.

Question > transportMode - this is a default value for a travel method, Which we don't really need. Under what circumstances would a survey offer 'default' answers for questions? Does it apply to the survey we are building?

Answer > This has been removed now.

Reply:
Hi @Amutha37,

I think you've misunderstood my reasoning for removing `transportIcon` and `transportMode` as parameter values for the `TravelMethodButtonsContainer` function. It has little to do with using `map` method.

just to be clear, this is the change I proposed:
![image](https://user-images.githubusercontent.com/4408259/157365279-23bf085b-22d0-4278-b292-7ef851fc9431.png)

my reasoning:
- parameters are 'variable' inputs - for values that would be used to calculate an output
- in the case of a `TravelMethodButtonsContainer` component, these travel method icons are the reason why this component exists; if you were able to pass in fruit icons instead, would it still be a `TravelMethodButtonsContainer` component?
- a similar argument can be made for `transportMode`

Functionally speaking `TravelMethodButtonsContainer` component will work the same way whether you decide to pass in `transportIcon` and `transportMode` as parameter values or with an `import` statement. It has more to do with the reasons 'why' one uses function parameters to obtain values, versus using imports to obtain values instead..

Reply:
I believe the advantage of having a tree structure for components is the possibility of making changes to each part of the page without affecting the page. However, on a side note, if we have a page which is working - we can always accept this solution for now and re-factor the code later one once the rest is done.

Reply:
The above question has been addressed . As at last commit. These 2 props doesn't make sense as a props so I have use them directly inside the button container.